### DISCUSSION TITLE
Reset rotary encoder counter?

### DISCUSSION THREAD

**Discussion by Bubbleman75** [arendst/Tasmota#13216] [Q&A] [answered]
Hi

Is there a way/a command to reset the encoder counter to zero (or set to value x), if a (end-) switch is triggered?
Something I can use in a rule?

THX

**Comment by Jason2866** [comment, marked as answer]
End switch with a rotary encoder?

**Reply by Bubbleman75** [reply]
It is a garagedoor opener project (Wemos D1 Mini). The encoder measures the doorposition (e.g. 30% open). There are two endswitches (full open,full close) I want to reset the encoderposition to zero every time the door is completely closed.

**Reply by sfromis** [reply]
Driver does not appear to have a reset capability, which makes sense as a rotary knob is generally assumed to be a direction and not a position.

For such a project, I'd consider skipping the rotary encoder driver, to instead use a simple pulse counter. This includes the possibility of resetting the counter.
